### PR TITLE
feature/8038-theo-BenefitsScreenEdgeCases

### PR DESCRIPTION
**Pull request by theodur**
## Description of Change

Adds logic for handling error and edge cases for Benefits category screen.

## Screenshots/Video

<details><summary>API failure</summary>Screenshot 2024-04-15 at 10 36 31 PM</details>

<details><summary>Availability framework access denied/content denied/allow function</summary>Screenshot 2024-04-15 at 10 37 10 PM</details>

<details><summary>Downtime</summary>Screenshot 2024-04-15 at 10 36 54 PM</details>

## Testing

- [ ] Tested on iOS 
- [ ] Tested on Android 

## Reviewer Validations

Benefits category screen reflect the logic and content in this [Mural](https://app.mural.co/t/adhoccorporateworkspace2583/m/adhoccorporateworkspace2583/1704381519703/cd6d78c5824a5fb0cc8b26597f0ad271072df8f2?sender=u4daa07e974be596b6b097258)

## PR Checklist

  **Reviewer:** Confirm the items below as you review

- [ ] PR is connected to issue(s)
- [ ] Tests are included to cover this change (when possible)
- [ ] No magic strings (All string unions follow the [Union -> Constant](https://github.com/department-of-veterans-affairs/va-mobile-app/blob/develop/VAMobile/src/constants/common.ts) type pattern)
- [ ] No secrets or API keys are checked in
- [ ] All imports are absolute (no relative imports)
- [ ] New functions and Redux work have proper TSDoc annotations

## For QA

[Run a build for this branch](https://github.com/department-of-veterans-affairs/va-mobile-app/actions/workflows/on_demand_build.yml)

### CODE REVIEW

File: VAMobile/src/screens/BenefitsScreen/BenefitsScreen.tsx [hunk 1]
```diff
@@ -34,12 +34,38 @@ function BenefitsScreen({}: BenefitsScreenProps) {
...
+  const appealsInDowntime = useDowntime(DowntimeFeatureTypeConstants.appeals)
+  const featureInDowntime = claimsInDowntime || appealsInDowntime
   const { data: userAuthorizedServices } = useAuthorizedServices({ enabled: screenContentAllowed('WG_Benefits') })
-  const { data: claimsData, isLoading: loadingClaimsAndAppeals } = useClaimsAndAppeals('ACTIVE', 1, {
+  const {
+    data: claimsAndAppeals,
+    isFetching: loadingClaimsAndAppeals,
+    isError: claimsAndAppealsError,
+  } = useClaimsAndAppeals('ACTIVE', 1, {
     enabled: (userAuthorizedServices?.claims || userAuthorizedServices?.appeals) && !claimsInDowntime,
```
Comment (dumathane) [inline VAMobile/src/screens/BenefitsScreen/BenefitsScreen.tsx:45]: Should this be...if they aren't BOTH in downtime?

Comment (dumathane) [reply VAMobile/src/screens/BenefitsScreen/BenefitsScreen.tsx:45]: Should potentially include the waygate check too?

Comment (dumathane) [reply VAMobile/src/screens/BenefitsScreen/BenefitsScreen.tsx:45]:
```suggestion
    enabled: ((userAuthorizedServices?.claims && !claimsInDowntime) || (userAuthorizedServices?.appeals && !appealsInDowntime)),
```

Comment (theodur) [reply VAMobile/src/screens/BenefitsScreen/BenefitsScreen.tsx]: Good point, added the waygate check and your suggested logic updates

**Review by dumathane — APPROVED** [review]
I'd like to see us remove all of the waygate code here. Availability Framework should be left to stand solo and not interact like this IMO and i'm working on that in a global thread. We'll see. I think at this point we're cool to pass it to Therese though.

**Comment by TKDickson** [comment]
Moved back to in-progress to make the changes with availability framework handling

**Review by dumathane — APPROVED** [review]

**Comment by TKDickson** [comment]
Need a max size on the icon, and the accessibility label still isn't reading out

**Pull request merged**